src/ruby/puppetserver-lib/puppet/server/compiler.rb

```diff
@@ -238,10 +238,18 @@ def get_facts_from_pdb(nodename, environment)
       # Initialize our server fact hash; we add these to each catalog, and they
       # won't change while we're running, so it's safe to cache the values.
       #
-      # This is lifted directly from Puppet::Indirector::Catalog::Compiler.
+      # See also set_server_facts in Puppet::Indirector::Catalog::Compiler in puppet.
       def set_server_facts
         @server_facts = {}
 
+        # Add our server Puppet Enterprise version, if available.
+        pe_version_file = '/opt/puppetlabs/server/pe_version'
+        if File.readable?(pe_version_file) and !File.zero?(pe_version_file)
+          pe_version_data = File.read(pe_version_file).chomp
+          pe_serverversion = pe_version_data.match(/(\d+\.\d+\.\d+)/)
```

Comment:
What else could be in the `pe_version_file`? I'm slightly wary of making assumptions about what the version could look like, since there was some talk of different versioning schemes (e.g. 2019.4 as opposed to 2019.4.0), though I bet those assumptions are EVERYWHERE..

Reply:
I haven't found a specification.
Certainly could loosen up the regex.

Reply:
I'm wondering if just reading the file is enough? Not sure what else might be in there.

Reply:
oh, I see this is literally exactly what's in puppet.. I guess we should probably keep them the same.

Reply:
It's a valid point. I'm open to dropping the match in both. 
I don't mind doing that work, if necessary.

Reply:
In puppet we generally try to parse version strings either using `Gem::Version.new` or `SemanticPuppet::Version.parse`. I don't know if one is a subset of the other? I assume both are available server-side? Might be good to switch both puppet and puppetserver.

Reply:
Perhaps we should just read the file, and defer to the code that writes the file in regard to the validity of its contents, since this code doesn't use the version itself?

```
    pe_version_file = '/opt/puppetlabs/server/pe_version'
    if File.readable?(pe_version_file) and !File.zero?(pe_version_file)
      @server_facts['pe_serverversion'] = File.read(pe_version_file).chomp
    end
```


I can easy add this to the commits for both puppet and here in puppetserver.

Reply:
yolo